lib/compiler/aro_translate_c.zig

```diff
@@ -215,80 +218,256 @@ fn prepopulateGlobalNameTable(c: *Context) !void {
+        // This type wasn't noticed by the name detection pass, so nothing has been treating this as
+        // a weak global name. We must mangle it to avoid conflicts with locals.
+        cur_name = try std.fmt.allocPrint(c.arena, "{s}_{d}", .{ want_name, c.getMangle() });
+    }
+
+    while (c.global_names.contains(cur_name)) {
+        cur_name = try std.fmt.allocPrint(c.arena, "{s}_{d}", .{ want_name, c.getMangle() });
+    }
+    return cur_name;
+}
+
+fn transRecordDecl(c: *Context, scope: *Scope, record_node: NodeIndex, field_nodes: []const NodeIndex) Error!void {
+    const node_types = c.tree.nodes.items(.ty);
+    const raw_record_ty = node_types[@intFromEnum(record_node)];
+    const record_decl = raw_record_ty.getRecord().?;
+    if (c.decl_table.get(@intFromPtr(record_decl))) |_|
+        return; // Avoid processing this decl twice
+    const toplevel = scope.id == .root;
+    const bs: *Scope.Block = if (!toplevel) try scope.findBlockScope(c) else undefined;
+
+    const container_kind: ZigTag = if (raw_record_ty.is(.@"union")) .@"union" else .@"struct";
+    const container_kind_name: []const u8 = @tagName(container_kind);
+
+    var is_unnamed = false;
+    var bare_name: []const u8 = c.mapper.lookup(record_decl.name);
+    var name = bare_name;
+
+    if (c.unnamed_typedefs.get(@intFromPtr(record_decl))) |typedef_name| {
+        bare_name = typedef_name;
+        name = typedef_name;
+    } else {
+        if (raw_record_ty.isAnonymousRecord(c.comp)) {
+            bare_name = try std.fmt.allocPrint(c.arena, "unnamed_{d}", .{c.getMangle()});
+            is_unnamed = true;
+        }
+        name = try std.fmt.allocPrint(c.arena, "{s}_{s}", .{ container_kind_name, bare_name });
+        if (toplevel and !is_unnamed) {
+            name = try mangleWeakGlobalName(c, name);
+        }
+    }
+    if (!toplevel) name = try bs.makeMangledName(c, name);
+    try c.decl_table.putNoClobber(c.gpa, @intFromPtr(record_decl), name);
+
+    const is_pub = toplevel and !is_unnamed;
+    const init_node = blk: {
+        var fields = try std.ArrayList(ast.Payload.Record.Field).initCapacity(c.gpa, record_decl.fields.len);
+        defer fields.deinit();
+
+        // TODO: Add support for flexible array field functions
+        var functions = std.ArrayList(ZigNode).init(c.gpa);
+        defer functions.deinit();
+
+        var unnamed_field_count: u32 = 0;
+
+        // If a record doesn't have any attributes that would affect the alignment and
+        // layout, then we can just use a simple `extern` type. If it does have attributes,
+        // then we need to inspect the layout and assign an `align` value for each field.
+        const has_alignment_attributes = record_decl.field_attributes != null or
+            raw_record_ty.hasAttribute(.@"packed") or
+            raw_record_ty.hasAttribute(.aligned);
+        const head_field_alignment: ?c_uint = headFieldAlignment(record_decl);
```

Comment:
`head_field_alignment` isn't used if `has_alignment_attributes` is false, so if that's the case we should just initialized it to null instead of computing a value via `headFieldAlignment`.